pkg/mimir/modules.go

```diff
@@ -829,11 +829,13 @@ func (t *Mimir) initStoreGateway() (serv services.Service, err error) {
 	t.Cfg.StoreGateway.ShardingRing.ListenPort = t.Cfg.Server.GRPCListenPort
 
 	// TODO: Remove in Mimir 2.12.
-	// Take value from Indexheader DeprecatedConfig if it is set
-	if t.Cfg.BlocksStorage.BucketStore.DeprecatedIndexHeaderLazyLoadingEnabled {
+	// Take the non-default value (default value is false) from IndexHeader deprecated config and if it is set, set it to the new config
+	if !t.Cfg.BlocksStorage.BucketStore.DeprecatedIndexHeaderLazyLoadingEnabled {
 		t.Cfg.BlocksStorage.BucketStore.IndexHeader.LazyLoadingEnabled = t.Cfg.BlocksStorage.BucketStore.DeprecatedIndexHeaderLazyLoadingEnabled
 	}
-	if t.Cfg.BlocksStorage.BucketStore.DeprecatedIndexHeaderLazyLoadingIdleTimeout > 0 {
+	// TODO: Remove in Mimir 2.12.
+	// Take the non-default value (default value is 1h) from IndexHeader deprecated config and if it is set, set it to the new config
+	if t.Cfg.BlocksStorage.BucketStore.DeprecatedIndexHeaderLazyLoadingIdleTimeout != 60*time.Minute {
```

Comment:
Should we move the `60m` value into a constant and reuse it in the flag registration? That will also make the comment redundant

Reply:
I tried to move to tsdb package but it result an import cycle. Let me find a good common package to place this.